go/enclave/core/batchlimiter.go

```diff
@@ -0,0 +1,107 @@
+package core
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+// BatchSizeLimiter - Acts as a limiter for batches based
+// the data from the transaction that we have to publish to the l1.
+// Acts as a calldata reservation system that accounts for both
+// transactions and cross chain messages.
+type batchSizeLimiter struct {
+	Size            uint64
```

Comment:
this needs a comment.
From what I can see, it's the remaining space in the batch.
So you should probably name it like that.